whylog/tests/tests_config/__init__.py

```diff
@@ -28,21 +47,38 @@ class TestBasic(TestCase):
     """
 
     def test_simple_transform(self):
-        parsers = {0: parser1, 1: parser2, 2: parser3}
-        effect_id = 2
-
-        constraint1 = UserConstraintIntent(identical_constr, [(0, 2), (1, 2)])
-        constraint2 = UserConstraintIntent(identical_constr, [(1, 3), (2, 2)])
-        constraint3 = UserConstraintIntent(different_constr, [(1, 2), (1, 3)])
-        constraint4 = UserConstraintIntent(
-            hetero_constr, [
-                (0, 3), (1, 4), (2, 4)
-            ], {"different": 1}
-        )
-        constraints = [constraint1, constraint2, constraint3, constraint4]
-
-        user_intent = UserRuleIntent(effect_id, parsers, constraints)
         rule = RegexRuleFactory.create_from_intent(user_intent)
 
         assert rule._effect.regex_str == regex3
         assert sorted(cause.regex_str for cause in rule._causes) == [regex1, regex2]
+
+    def test_parser_serialization(self):
+        parser1 = RegexParserFactory.create_from_intent(parser_intent1)
+        parser2 = RegexParserFactory.create_from_intent(parser_intent2)
+        parser3 = RegexParserFactory.create_from_intent(parser_intent3)
+        parsers_list = [parser1, parser2, parser3]
+
+        parsers_dao_list = [parser.serialize() for parser in parsers_list]
+        dumped_parsers = yaml.dump_all(parsers_dao_list, explicit_start=True)
+        loaded_parsers = [
+            RegexParserFactory.deserialize(dumped_parser)
```

Comment:
This line suggests that `RegexParserFactory` objects are being deserialized. Either deserialize `RegexParser` objects or `RegexParserFactory.from_dao()`